### PR TITLE
Fix dataset filename

### PR DESCRIPTION
**Pull request by twd2**
#335

### CODE REVIEW

**Comment by breeswish** [comment]
You can use [mimetypes](https://docs.python.org/3/library/mimetypes.html)

File: vj4/handler/contest.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ class ContestCodeHandler(base.OperationHandler):
       zfile.create_system = 0
     zip_file.close()
 
-    await self.binary(output_buffer.getvalue(), 'application/zip')
+    await self.binary(output_buffer.getvalue(), 'application/zip', file_name='code.zip')
```
Comment (breeswish) [inline vj4/handler/contest.py]: use contest name

File: vj4/handler/fs.py [hunk 2]
```diff
@@ -25,10 +25,15 @@ class FsGetHandler(base.Handler):
     self.response.content_type = grid_out.content_type or 'application/octet-stream'
     # FIXME(iceboy): For some reason setting response.content_length doesn't work in aiohttp 2.0.6.
     self.response.headers['Content-Length'] = str(grid_out.length)
+    ext = mimetypes.guess_extension(self.response.content_type)
+    if not ext:
+      ext = ''
+    self.response.headers.add('Content-Disposition',
```
Comment (breeswish) [inline vj4/handler/fs.py]: Can we reuse handler.binary?

Comment (twd2) [reply vj4/handler/fs.py]: No, because this contains cache-control, block write, ...

Comment (breeswish) [reply vj4/handler/fs.py]: ok

**Pull request merged**